Source/Evolve/WarpXComputeDt.cpp

```diff
@@ -27,29 +28,33 @@
 #include <algorithm>
 #include <memory>
 
+AMREX_FORCE_INLINE amrex::Real
+minDim (const amrex::Real* x)
+{
+#if defined(WARPX_DIM_1D_Z)
+    return x[0];
+#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+    return std::min(x[0], x[1]);
+#else
+    return std::min(x[0], std::min(x[1], x[2]));
+#endif
+}
+
 /**
  * Determine the timestep of the simulation. */
 void
 WarpX::ComputeDt ()
 {
     // Handle cases where the timestep is not limited by the speed of light
-    if (electromagnetic_solver_id == ElectromagneticSolverAlgo::None ||
```

Comment:
A comment here, is that when `electromagnetic_solver_id == ElectromagneticSolverAlgo::None` it does not necessarily mean that the electrostatic solver is being used. There are cases when no solver is being used. But that should be fine since the new CFL limit should work for those cases as well.

Would this CFL limit also work for the hybrid solver? @roelof-groenewald

Reply:
This should work for the hybrid solvers used in my field (plasma propulsion), as the condition I impose is typically used to set a constant global timestep. As long as the way the leapfrog algorithm interacts with the adaptive timestep maintains its accuracy and consistency, I don't think there should be any problems. The only issues would stem from how (if at all) the Ohm's law solve interacts with the timestep.